src/main/java/com/redhat/red/build/koji/KojiClientHelper.java

```diff
@@ -0,0 +1,76 @@
+package com.redhat.red.build.koji;
+
+import com.redhat.red.build.koji.model.xmlrpc.KojiArchiveInfo;
+import com.redhat.red.build.koji.model.xmlrpc.KojiArchiveQuery;
+import com.redhat.red.build.koji.model.xmlrpc.KojiMultiCallValueObj;
+import com.redhat.red.build.koji.model.xmlrpc.KojiSessionInfo;
+import com.redhat.red.build.koji.model.xmlrpc.messages.MultiCallRequest;
+import com.redhat.red.build.koji.model.xmlrpc.messages.MultiCallResponse;
+import org.commonjava.rwx.core.Registry;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.redhat.red.build.koji.model.xmlrpc.messages.Constants.LIST_ARCHIVES;
+import static com.redhat.red.build.koji.model.xmlrpc.messages.MultiCallRequest.getBuilder;
+
+/**
+ * This class provides advanced helper methods, e.g., using multicall to batch query operations.
+ */
+public class KojiClientHelper
+{
+    private final KojiClient client;
+
+    private final Registry registry = Registry.getInstance();
+
+    public KojiClientHelper( KojiClient client )
+    {
+        this.client = client;
+    }
+
+    /**
+     * This method sends a list of KojiArchiveQuery queries, and get back the responses for those queries.
+     * The responses are in a List and have the same order corresponding to the query objects,
+     * i.e., if the requests contains [req-1, req-2, ...], the responses are [resp-1, resp-2, ...].
+     *
+     * If any query gets no result, its response would be an empty List.
+     *
+     * @param queries a list of KojiArchiveQuery queries
+     * @param session
+     * @return a list of response objects. Each response object is a List of KojiArchiveInfo instance.
+     * @throws KojiClientException
+     */
+    public List<List<KojiArchiveInfo>> listArchives( List<KojiArchiveQuery> queries, KojiSessionInfo session )
+                    throws KojiClientException
```

Comment:
We actually have at least five different query types:

* `KojiArchiveQuery`
* `KojiBuildQuery`
* `KojiBuildTypeQuery`
* `KojiPackageQuery`
* `KojiTagQuery`

All of these do happen to inherit from the `KojiQuery` base type, although unfortunately their `Request` objects do not. It would be nice to support other query types without repeating the same code.

Reply:
I believe this is doable. We just need to pass a req type and resp type to the method to render/parse content. I will make an update soon.